cfpb/cfgov-refresh
# Add Spanish translations to Financial Well-Being Scale Django app

This adds an iteration on the Django app version of the Financial Well-Being Scale (https://www.consumerfinance.gov/consumer-tools/financial-well-being/) to include its Spanish version.

## Additions

- A `TranslatedTemplateView` view to direct users to the proper URL for their chosen language 
- Links for users to switch between English and Spanish on most pages of the app
- Spanish translations for the Financial Well-Being Scale added to `django.po`

## Testing

1. Load this branch on your machine, and visit http://localhost:8000/consumer-tools/financial-well-being/
2. Fill out the entire questionnaire, checking that the JavaScript and data flow for the default English are working as expected. Check the `/financial-well-being/about/` and `/financial-well-being/error/` slugs as well.
3. Return to the start of the questionnaire, and click the link under the hero to view the page in Spanish: "Ver página en español". Fill out the entire questionnaire in Spanish, checking that the JavaScript and data flow for Spanish are working as expected. Check the `/financial-well-being/about/es/` and `/financial-well-being/error/es/` slugs as well.
4. Return to the start of the questionnaire, and click the link under the hero to view the page in English again: "View page in English".
5. Manually add `/es/` to the end of the default start page for the tool. Check that the page still switches to the Spanish translation.

## Screenshots

### Start page

<img width="1187" alt="fwb-home-english" src="https://user-images.githubusercontent.com/1183435/79508489-a899ce80-8007-11ea-99ae-528721ea3b8d.png">

<img width="1186" alt="fwb-home-spanish" src="https://user-images.githubusercontent.com/1183435/79508501-adf71900-8007-11ea-8f27-f1e805180dcc.png">

### Results page

<img width="1165" alt="fwb-results-english" src="https://user-images.githubusercontent.com/1183435/79508520-b6e7ea80-8007-11ea-9add-76d568acecde.png">

<img width="1118" alt="fwb-results-spanish" src="https://user-images.githubusercontent.com/1183435/79508538-bd766200-8007-11ea-86f5-31d43308412a.png">

### About page

<img width="1159" alt="fwb-about-english" src="https://user-images.githubusercontent.com/1183435/79508558-c404d980-8007-11ea-9d13-125a2f4f74e4.png">

<img width="1169" alt="fwb-about-spanish" src="https://user-images.githubusercontent.com/1183435/79508569-c9622400-8007-11ea-928b-47c82593727f.png">

### Error page

<img width="862" alt="fwb-error-english" src="https://user-images.githubusercontent.com/1183435/79508579-cebf6e80-8007-11ea-9c3a-b7dae3441558.png">

<img width="849" alt="fwb-error-spanish" src="https://user-images.githubusercontent.com/1183435/79508590-d252f580-8007-11ea-8ca2-7be44e8f4a5c.png">

## Notes

- There isn't a link to switch between English and Spanish on the `/results/` page, as the user's data doesn't flow through a change in the views. You can see what would happen by manually adding `/es/` to the end of the URL after filling out the questionnaire.

## Checklist


## Testing checklist

### Browsers


### Other
